tests/cases/conformance/enums/enumLiteralBasics.ts

```diff
@@ -0,0 +1,38 @@
+// Enum without initializers have first member = 0 and successive members = N + 1
+
+// Enum literal syntax does not implement auto-incrementing behaviour.
+let ExistingShorthand = "exists";
+const E1: enum = {
+    NonexistingShorthand, // error -- EnumLiteralExpressions require explicit property definitions.
+    ExistingShorthand, // error -- EnumLiteralExpressions require explicit property definitions.
+    Int: 1, // ok
+    String: "string", // ok
+    Flag: 8, // ok
+};
+
+// Valid assignments
+const nonexist: E1 = E1.NonexistingShorthand; // ok
+const exist: E1 = E1.ExistingShorthand; // ok
+const ival: E1 = E1.Int; // ok
+const sval: E1 = E1.String; // ok
```

Comment:
Does this implementation support types for individual variants? In other words, would the following work?

```
const exist: E1.ExistingShorthand = E1.ExistingShorthand; // ok
const ival: E1.Int = E1.Int; // ok
const sval: E1.String = E1.String;
```

I looked through the PR, but did not saw code handling it or tests for this. It seems however to delegate to the existing enum handling in some places so I assume that it should work. I would appreciate a confirmation, thank you.

Reply:
I've made an effort to add a few of these tests (and they seem to be working, although it is a bit loose with accepting a wider number type for a unique numeric enum value. I believe that's normal for TS enums, though), but mostly I was focused on getting the invariant checks working. Thanks for the suggestion.

Reply:
Thank you :slightly_smiling_face: